Q1. Pearson correlation coefficient is a measure of the linear relationship between two variables. Suppose
you have collected data on the amount of time students spend studying for an exam and their final exam
scores. Calculate the Pearson correlation coefficient between these two variables and interpret the result. 


To calculate the Pearson correlation coefficient between the amount of time students spend studying for an exam and their final exam scores, you would need a dataset with paired observations for both variables.

Assuming you have such a dataset, follow these steps to calculate the Pearson correlation coefficient:

Calculate the mean (average) of the study time variable and the mean of the exam scores variable.
Calculate the difference between each study time observation and the mean study time.
Calculate the difference between each exam score observation and the mean exam score.
Multiply each pair of differences obtained in steps 2 and 3.
Sum up all the products obtained in step 4.
Calculate the square root of the sum of squared differences for the study time variable and the sum of squared differences for the exam scores variable.
Divide the sum of products obtained in step 5 by the product of the square roots calculated in step 6.
The resulting value is the Pearson correlation coefficient, typically denoted by the symbol "r."
Interpreting the Pearson correlation coefficient:

The Pearson correlation coefficient, denoted by "r," ranges from -1 to 1. It represents the strength and direction of the linear relationship between the two variables.

If the correlation coefficient is close to 1, it indicates a strong positive linear relationship. In this context, it would mean that there is a strong positive correlation between the amount of time students spend studying and their final exam scores. Higher study time would tend to be associated with higher exam scores.

If the correlation coefficient is close to -1, it indicates a strong negative linear relationship. In this case, it would mean that there is a strong negative correlation between the amount of time students spend studying and their final exam scores. Higher study time would tend to be associated with lower exam scores.

If the correlation coefficient is close to 0, it indicates a weak or no linear relationship. This means that the study time and exam scores are not strongly associated with each other in a linear manner. 

Q2. Spearman's rank correlation is a measure of the monotonic relationship between two variables.
Suppose you have collected data on the amount of sleep individuals get each night and their overall job
satisfaction level on a scale of 1 to 10. Calculate the Spearman's rank correlation between these two
variables and interpret the result. 

To calculate Spearman's rank correlation between the amount of sleep individuals get each night and their overall job satisfaction level, you would need a dataset with paired observations for both variables.

Assuming you have such a dataset, follow these steps to calculate Spearman's rank correlation:

Rank the values of both variables separately from lowest to highest. If there are ties, assign the average rank to the tied observations.

Calculate the difference between the ranks for each paired observation.

Square the differences obtained in step 2.

Sum up all the squared differences.

Calculate the Spearman's rank correlation coefficient, denoted by the symbol "ρ," using the formula:

ρ = 1 - (6 * sum of squared differences) / (n * (n^2 - 1))

where n is the number of paired observations.

Interpreting the Spearman's rank correlation coefficient:

The Spearman's rank correlation coefficient, denoted by "ρ," ranges from -1 to 1. It measures the strength and direction of the monotonic relationship between the two variables.

If the correlation coefficient is close to 1, it indicates a strong positive monotonic relationship. In this context, it would mean that as the amount of sleep individuals get each night increases, their overall job satisfaction level tends to increase as well. This suggests a positive association between the two variables.

If the correlation coefficient is close to -1, it indicates a strong negative monotonic relationship. It would mean that as the amount of sleep individuals get each night increases, their overall job satisfaction level tends to decrease. This suggests a negative association between the two variables.



Q3. Suppose you are conducting a study to examine the relationship between the number of hours of
exercise per week and body mass index (BMI) in a sample of adults. You collected data on both variables
for 50 participants. Calculate the Pearson correlation coefficient and the Spearman's rank correlation
between these two variables and compare the results. 

To calculate the Pearson correlation coefficient and the Spearman's rank correlation coefficient between the number of hours of exercise per week and body mass index (BMI) in a sample of 50 participants, you would need a dataset with paired observations for both variables.

Assuming you have such a dataset, here's how you can calculate the two correlation coefficients:

Calculate the Pearson correlation coefficient (r):

Compute the mean (average) of the exercise hours variable (X̄) and the mean of the BMI variable (Ȳ).
Calculate the difference between each exercise hours observation (X) and the mean exercise hours (X - X̄).
Calculate the difference between each BMI observation (Y) and the mean BMI (Y - Ȳ).
Multiply each pair of differences obtained in the previous step [(X - X̄) * (Y - Ȳ)].
Sum up all the products obtained in the previous step.
Calculate the square root of the sum of squared differences for the exercise hours variable and the sum of squared differences for the BMI variable.
Divide the sum of products obtained earlier by the product of the square roots calculated in the previous step.
The resulting value is the Pearson correlation coefficient (r).
Calculate the Spearman's rank correlation coefficient (ρ):

Rank the values of both variables separately from lowest to highest. If there are ties, assign the average rank to the tied observations.
Calculate the difference between the ranks for each paired observation.
Square the differences obtained in the previous step.
Sum up all the squared differences.
Calculate the Spearman's rank correlation coefficient using the formula:
ρ = 1 - (6 * sum of squared differences) / (n * (n^2 - 1))
where n is the number of paired observations.


Q4. A researcher is interested in examining the relationship between the number of hours individuals
spend watching television per day and their level of physical activity. The researcher collected data on
both variables from a sample of 50 participants. Calculate the Pearson correlation coefficient between
these two variables. 

To calculate the Pearson correlation coefficient between the number of hours individuals spend watching television per day and their level of physical activity in a sample of 50 participants, you would need a dataset with paired observations for both variables.

Assuming you have such a dataset, follow these steps to calculate the Pearson correlation coefficient:

Compute the mean (average) of the television viewing hours variable (X̄) and the mean of the physical activity level variable (Ȳ).
Calculate the difference between each television viewing hours observation (X) and the mean television viewing hours (X - X̄).
Calculate the difference between each physical activity level observation (Y) and the mean physical activity level (Y - Ȳ).
Multiply each pair of differences obtained in the previous step [(X - X̄) * (Y - Ȳ)].
Sum up all the products obtained in the previous step.
Calculate the square root of the sum of squared differences for the television viewing hours variable and the sum of squared differences for the physical activity level variable.
Divide the sum of products obtained earlier by the product of the square roots calculated in the previous step.
The resulting value is the Pearson correlation coefficient (r).


Q5. A survey was conducted to examine the realtionship between age and preference fo ra particular brand of soft drink. The survey result are shown below :    

In [6]:
import scipy.stats as stats

#
age = [25, 42, 37, 19, 31, 28]
soft_drink_preference = ['Coke', 'Pepsi', 'Mountain Dew', 'Coke', 'Pepsi', 'Coke']


mapping = {'Coke': 1, 'Pepsi': 2, 'Mountain Dew': 3}
numerical_preference = [mapping[p] for p in soft_drink_preference]


correlation_coefficient, _ = stats.pearsonr(age, numerical_preference)


print("Pearson correlation coefficient:", correlation_coefficient)


Pearson correlation coefficient: 0.7587035441865057


Q6. A company is interested in examining the relationship between the number of sales calls made per day
and the number of sales made per week. The company collected data on both variables from a sample of
30 sales representatives. Calculate the Pearson correlation coefficient between these two variables.

In [1]:
# Sample data
sales_calls = [10, 12, 8, 15, 6, 9, 14, 7, 11, 5, 13, 16, 9, 8, 10, 17, 4, 12, 6, 15, 8, 11, 10, 7, 9, 14, 13, 5, 12, 16]
sales_week = [7, 9, 4, 12, 3, 5, 11, 2, 8, 1, 10, 15, 6, 4, 8, 14, 1, 7, 3, 12, 5, 9, 7, 3, 4, 11, 8, 2, 7, 13]

# Calculate Pearson correlation coefficient
import numpy as np
correlation = np.corrcoef(sales_calls, sales_week)[0, 1]

print("Pearson correlation coefficient:", correlation)

# Interpret the correlation coefficient
# Strong positive correlation: 0.7 to 1.0
# Weak positive correlation: 0.3 to 0.7
# No correlation: 0.0 to 0.3
# Weak negative correlation: -0.3 to -0.7
# Strong negative correlation: -0.7 to -1.0

if abs(correlation) >= 0.7:
    if correlation > 0:
        print("There is a strong positive correlation between sales calls and sales.")
    else:
        print("There is a strong negative correlation between sales calls and sales.")
elif abs(correlation) >= 0.3:
    if correlation > 0:
        print("There is a weak positive correlation between sales calls and sales.")
    else:
        print("There is a weak negative correlation between sales calls and sales.")
else:
    print("There is no significant correlation between sales calls and sales.")

Pearson correlation coefficient: 0.9718052467724818
There is a strong positive correlation between sales calls and sales.


the Pearson correlation coefficient between the number of sales calls made per day and the number of sales made per week is 0.972, which indicates a strong positive correlation between the two variables. In other words, there is a statistically significant relationship between the number of sales calls made and the number of sales achieved. As the number of sales calls increases, the number of sales made also tends to increase.